src/node_nanomsg.cc

```diff
@@ -191,35 +190,40 @@ NAN_METHOD(Err) {
   info.GetReturnValue().Set(Nan::New(nn_strerror(nn_errno())).ToLocalChecked());
 }
 
-typedef struct nanomsg_socket_s {
-  uv_poll_t poll_handle;
-  uv_os_sock_t sockfd;
-  Nan::Callback *callback;
-} nanomsg_socket_t;
+class NanoMsgPollCtx {
+  private:
+  public:
+    uv_poll_t poll_handle;
+    uv_os_sock_t sockfd;
+    Nan::Callback *callback;
+    NanoMsgPollCtx (int s, bool is_sender, Local<v8::Function> cb) {
+      callback = new Nan::Callback(cb);
+      size_t siz = sizeof(uv_os_sock_t);
+      nn_getsockopt(s, NN_SOL_SOCKET, is_sender ? NN_SNDFD : NN_RCVFD, &sockfd,
+          &siz);
+      poll_handle.data = this;
+    }
+    ~NanoMsgPollCtx () {
+      delete callback;
+    }
```

Comment:
No, that would not work (since you could not set the member to the previously constructed callback.

Accessor lists work as such:
```c++
class Foo {
 public:
  Foo(int n) : n_(n) {}

 private:
    int n_;
};

class Bar {
 public:
  Bar(Foo f) : f_(f) {}

 private:
  Foo f_;
};
```
In `Bar(int n) : f_(n) {}`, read the `f_(n)` part as `Foo f_(n)`, e.g. same as in
```c++
int main() {
  int n = 4;
  Foo f_(n);

  return 0;
}